rfd/0135-kube-apps-discovery.md

```diff
@@ -0,0 +1,189 @@
+---
+    authors: Anton Miniailo (anton@goteleport.com)
+    state: draft
+---
+
+# RFD 135 - Kubernetes App discovery
+
+## Required Approvers
+
+- Engineering: `@r0mant` && `@tigrato`
+- Product: `@klizhentas` || `@xinding33`
+- Security: `@reedloden` || `@jentfoo`
+
+## What
+
+Proposes discovery of apps that are hosted inside Kubernetes clusters and exposed as services.
+
+Related issue: [#25538](https://github.com/gravitational/teleport/issues/25538)
+
+## Why
+
+Some users dynamically deploy apps into their Kubernetes clusters and expose those as Kubernetes services. When they actively
+create/update/remove apps, manually trying to mirror these changes to Teleport may be time-consuming and prone to
+errors, especially for organizations that have a large number of services running in their Kubernetes clusters.
+With the changes described in this RFD, Teleport will be able to discover Kubernetes service and automatically register them as
+Teleport apps and keep them in up to date state, greatly simplifying process and making user experience much better.
+
+## Scope
+
+In this document we focus on dynamically creating Teleport apps based on discovery of services in Kubernetes clusters that are running
+Teleport agents. Discovery service and app service should be running inside the Kubernetes cluster whose services are being exposed.
+
+## Details
+
+Kubernetes app discovery will build upon existing infrastructure for other discovery capabilities that are already there. We will add another
+type of matchers to the discovery service configuration - `kubernetes_apps`. When discovery service sees this, it will periodically
```

Comment:
Yep, I think that looks good 👍 Nice way to plan for extension to kube DB discovery

Discovery group as cluster name is an option. I think it's a bit less clear, but more straightforward than specifying same cluster name for every matcher 🤔  We could also add top-level config `kubernetes_cluster` that would control it. If it's missing it's equal to discovery group value, but if it present we use it. And at least one of the two fields (`discovery_group` or `kubernetes_cluster`) should be set. I think in general `discovery_group` should be mandatory, but this is regardless of current RFD and it might require some backwards compatibility thoughts.